pkg/cli/cli.go

```diff
@@ -113,10 +121,10 @@ func (c cli) Run() error {
 	return c.cmd.Execute()
 }
 
-// WithCommandName is an Option that sets the cli's root command name.
-func WithCommandName(name string) Option {
+// WithRootCommandConfig configures a CLI's root command.
+func WithRootCommandConfig(cfg RootCommandConfig) Option {
 	return func(c *cli) error {
-		c.commandName = name
+		c.cmdCfg = cfg
```

Comment:
You didn't apply this. Providing a partial struct will overwrite with empty strings other fields.

Reply:
Yes, that's intentional, as I am setting the default for the other fields in `defaultCommand()` if they are empty after calling the `WithRootCommandConfig(cfg)` option.

I think it will do the same work as intended by this suggestion but also not set these descriptor fields early, so that we don't lose track of them when using them at the end., as per an earlier suggestion of yours. 😅